app/assets/html/sidebar.html

```diff
@@ -4,7 +4,7 @@
         <ul>
             <li ng-repeat="gladiatorId in gladiatorIds">
                 <div class="gladiator-heading-container menu-item" ng-click="toggleGladiatorPanel(gladiatorId)">
-                    <img class="user-avatar" ng-src="{{gladiatorById[gladiatorId].avatar}}"></img>
+                    <img class="user-avatar" ng-click="goToProfile(gladiatorId); $event.stopPropagation()" ng-src="{{gladiatorById[gladiatorId].avatar}}"></img>
```

Comment:
I feel you're mixing lots of features into one bundle. You really should break them up.